src/components/AuthorViewpoint/ViewpointCard.tsx

```diff
@@ -0,0 +1,134 @@
+"use client";
+import { TrashIcon, PlusIcon } from "@heroicons/react/24/outline";
+import { Button, TextInput } from "@mantine/core";
+import { useState, useRef, useEffect } from "react";
+import Link from "next/link";
+
+type ViewpointCardProps = {
+    issueId: string;
+    viewpointTitle: string;
+    setViewpointTitle: (value: string) => void;
+    setViewpointContent: (value: string) => void;
+    publishViewpoint: () => void;
+};
+
+export default function ViewpointCard({
+    issueId,
+    viewpointTitle,
+    setViewpointTitle,
+    setViewpointContent,
+    publishViewpoint,
+}: ViewpointCardProps) {
+    const [contentEmpty, setContentEmpty] = useState<boolean>(true);
+    const inputRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        //manage the placeholder in the content area
+        if (inputRef?.current === null || inputRef.current.innerHTML !== "")
+            return;
+        const placeholderElement = document.createElement("p");
+        placeholderElement.className = "text-neutral-500";
+        placeholderElement.textContent =
+            "開始打字，或選取一段文字來新增引注資料";
+        inputRef.current.appendChild(placeholderElement);
+    }, [inputRef]);
+
+    const onPublish = () => {
+        if (viewpointTitle == "" || contentEmpty) {
+            console.log("Title and Content MUST NOT be empty");
```

Comment:
Consider using a toast to display this message to the user as most users won't check the JS console

Reply:
Furthermore, if the message were to appear in a toast, please translate the message into Chinese.

Reply:
The console is added to check if the logic works when developing.